cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/KafkaBrokerConfigurationDiffTest.java

```diff
@@ -0,0 +1,256 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+
+package io.strimzi.operator.cluster.operator.resource;
+
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import io.strimzi.operator.cluster.KafkaVersionTestUtils;
+import io.strimzi.operator.cluster.model.KafkaVersion;
+import io.strimzi.test.TestUtils;
+import org.apache.kafka.clients.admin.Config;
+import org.apache.kafka.clients.admin.ConfigEntry;
+import org.apache.kafka.common.config.ConfigResource;
+import org.junit.jupiter.api.Test;
+
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.CoreMatchers.is;
+
+public class KafkaBrokerConfigurationDiffTest {
+
+    private static final KafkaVersion.Lookup VERSIONS = KafkaVersionTestUtils.getKafkaVersionLookup();
+    private static final String KAFKA_VERSION = "2.4.0";
+    KafkaVersion kafkaVersion = VERSIONS.version(KAFKA_VERSION);
+    private int brokerId = 0;
+
+    public ConfigMap getTestingDesiredConfiguration(ArrayList<ConfigEntry> additional) {
+        InputStream is = getClass().getClassLoader().getResourceAsStream("desired-kafka-broker.conf");
+        String desiredConfigString = TestUtils.readResource(is);
+
+        for (ConfigEntry ce: additional) {
+            desiredConfigString += "\n" + ce.name() + "=" + ce.value();
+        }
+
+        ConfigMap configMap = new ConfigMap();
+
+        HashMap<String, String> data = new HashMap();
+        data.put("server.config", desiredConfigString);
+        configMap.setData(data);
+        return configMap;
+    }
+
+    public Map<ConfigResource, Config> getTestingCurrentConfiguration(ArrayList<ConfigEntry> additional) {
+        Map<ConfigResource, Config> current = new HashMap<>();
+        ConfigResource cr = new ConfigResource(ConfigResource.Type.BROKER, "0");
+        InputStream is = getClass().getClassLoader().getResourceAsStream("current-kafka-broker.conf");
+
+        List<String> configList = Arrays.asList(TestUtils.readResource(is).split(System.getProperty("line.separator")));
```

Comment:
How does this work when the JVM OS line separator differs from the separator used in the conf file?

Reply:
Did not test :(